transmogrifier/sources/transformer.py

```diff
@@ -153,233 +208,308 @@ def parse_source_file(cls, source_file: str) -> Iterator[JSON | Tag]:
+            source_record_id: Record identifier for the source record.
             source_record: A single source record.
         """
-        return {}
+        pass
 
+    @classmethod
     @abstractmethod
-    def transform(self, source_record: JSON | Tag) -> Optional[TimdexRecord]:
+    def get_source_record_id(cls, source_record: JSON | Tag) -> str:
         """
-        Transform a source record into a TIMDEX record.
+        Get or generate a source record ID from a source record.
 
-        Must be overridden by format subclasses.
+        Must be overridden by source subclasses.
 
         Args:
             source_record: A single source record.
         """
-        return None
+        pass
 
     @classmethod
     @abstractmethod
-    def get_valid_title(cls, source_record_id: str, source_record: JSON | Tag) -> str:
+    def record_is_deleted(cls, source_record: JSON | Tag) -> bool:
         """
-        Retrieves main title(s) from a source record and returns a valid title string.
+        Determine whether record has a status of deleted.
 
         Must be overridden by source subclasses.
 
         Args:
-            source_record_id: Record identifier for the source record.
             source_record: A single source record.
         """
-        return ""
+        pass
 
-    @classmethod
-    @abstractmethod
-    def get_source_link(
-        cls, source_base_url: str, source_record_id: str, source_record: JSON | Tag
-    ) -> str:
+    def get_optional_fields(self, source_record: JSON | Tag) -> Optional[dict]:
         """
-        Class method to set the source link for the item.
+        Retrieve optional TIMDEX fields from a source record.
 
-        Must be overridden by source subclasses.
+        May be overridden by source subclasses.
 
         Args:
-            source_base_url: Source base URL.
-            source_record_id: Record identifier for the source record.
             source_record: A single source record.
         """
-        return ""
+        return {}
 
+
+class JsonTransformer(Transformer):
+    """JSON transformer class."""
+
+    @final
     @classmethod
-    @abstractmethod
-    def get_timdex_record_id(
-        cls, source: str, source_record_id: str, source_record: Tag
-    ) -> str:
+    def parse_source_file(cls, source_file: str) -> Iterator[JSON]:
         """
-        Class method to set the TIMDEX record id.
+        Parse JSON file and return source records as JSON objects via an iterator.
 
-        Must be overridden by source subclasses.
+        May not be overridden.
 
         Args:
-            source: Source name.
-            source_record_id: Record identifier for the source record.
-            source_record: A single source record.
+            source_file: A file containing source records to be transformed.
         """
-        return ""
+        with open(source_file, "rb") as file:
+            for record in json.load(file):
+                yield record
```

Comment:
Don't think this is required now, but noting for the future: I continue to think it might be nice if Transmogrifier could handle [JSON Lines](https://jsonlines.org/) as an input file type.  

This would allow "harvesters" (broadly speaking) to write JSON Lines files for Transmog to work with.

Again, not required, but noting that it could be in this `parse_source_file` where that functionality is introduced later.

Reply:
Noted, I'll save that for later!